akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -263,9 +266,14 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
    * An actor system can only join a cluster once. Additional attempts will be ignored.
    * When it has successfully joined it must be restarted to be able to join another
    * cluster or to join the same cluster again.
+   * @throws TerminatedClusterException(message:String) Throws when the cluster have been shutdown.
    */
-  def join(address: Address): Unit =
-    clusterCore ! ClusterUserAction.JoinTo(address)
+  def join(address: Address): Unit = {
+    if (_isTerminated.get())
+      throw new TerminatedClusterException("Cluster have been shutdown,you must restart the actor system to rejoin the cluster.")
```

Comment:
I'm completely convinced that we should add this. Those commands are not guaranteed to succeed in any case and they are intended to be idempotent for safe retry.